pkg/parse/parse.go

```diff
@@ -612,19 +636,28 @@ func PlatformFromOptions(c *cobra.Command) (os, arch string, err error) {
 			arch = selectedArch
 		}
 	}
-
+	if c.Flag("variant").Changed {
+		if selectedVariant, err := c.Flags().GetString("variant"); err == nil {
+			variant = selectedVariant
+		}
+	}
+	platforms = []struct{ OS, Arch, Variant string }{{os, arch, variant}}
+	var platformSpecs []string
 	if c.Flag("platform").Changed {
-		if pf, err := c.Flags().GetString("platform"); err == nil {
-			selectedOS, selectedArch, _, err := Platform(pf)
-			if err != nil {
-				return "", "", errors.Wrap(err, "unable to parse platform")
-			}
-			arch = selectedArch
-			os = selectedOS
+		platforms = nil
+		platformSpecs, err = c.Flags().GetStringSlice("platform")
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to parse platform")
 		}
 	}
-
-	return os, arch, nil
+	for _, pf := range platformSpecs {
```

Comment:
This could also be inside the `("platform").Changed` conditional.

Entirely up to you.